app/src/main/java/app/revanced/manager/ui/screen/settings/update/UpdateScreen.kt

```diff
@@ -0,0 +1,224 @@
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import app.revanced.manager.BuildConfig
+import app.revanced.manager.R
+import app.revanced.manager.ui.component.AppTopBar
+import app.revanced.manager.ui.component.settings.Changelog
+import app.revanced.manager.ui.viewmodel.UpdateViewModel
+import app.revanced.manager.ui.viewmodel.UpdateViewModel.Changelog
+import app.revanced.manager.ui.viewmodel.UpdateViewModel.State
+import app.revanced.manager.util.formatNumber
+import app.revanced.manager.util.relativeTime
+import com.gigamole.composefadingedges.content.FadingEdgesContentType
+import com.gigamole.composefadingedges.content.scrollconfig.FadingEdgesScrollConfig
+import com.gigamole.composefadingedges.fill.FadingEdgesFillType
+import com.gigamole.composefadingedges.verticalFadingEdges
+import org.koin.androidx.compose.getViewModel
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+@Stable
+fun UpdateScreen(
+    onBackClick: () -> Unit,
+    vm: UpdateViewModel = getViewModel()
+) {
+    Scaffold(
+        topBar = {
+            AppTopBar(
+                title = stringResource(R.string.updates),
+                onBackClick = onBackClick
+            )
+        }
+    ) { paddingValues ->
+        AnimatedVisibility(visible = vm.showInternetCheckDialog) {
+            meteredDownloadConfirmationDialog(vm)
+        }
+        Column(
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(paddingValues)
+                .padding(vertical = 16.dp, horizontal = 24.dp)
+                .verticalScroll(rememberScrollState()),
+            verticalArrangement = Arrangement.spacedBy(32.dp)
+        ) {
+            Header(vm)
+            vm.changelog?.let { changelog ->
+                Divider()
+                Changelog(changelog)
+            } ?: Spacer(modifier = Modifier.weight(1f))
+            Buttons(vm, onBackClick)
+        }
+    }
+}
+
+@Composable
+private fun meteredDownloadConfirmationDialog(vm: UpdateViewModel) {
+    AlertDialog(
+        onDismissRequest = { vm.showInternetCheckDialog = false },
+        dismissButton = {
+            TextButton(
+                onClick = {
+                    vm.showInternetCheckDialog = false
+                }
+            ) {
+                Text(stringResource(id = R.string.cancel))
+            }
+        },
+        confirmButton = {
+            TextButton(
+                onClick = {
+                    vm.showInternetCheckDialog = false
+                    vm.downloadUpdate(true)
+                }
+            ) {
+                Text(stringResource(id = R.string.download))
+            }
+        },
+        title = {
+            Text(text = stringResource(id = R.string.download_update_confirmation))
+        },
+        icon = {
+            Icon(Icons.Outlined.Update, stringResource(id = R.string.update))
```

Comment:
The description is unnecessary since the title **should** convey enough information

Reply:
A lot of dialogs can have their purpose conveyed by the title, a description is an addition to go a little more in-depth

Reply:
Yes, but decorative icons don't need a text replacement

Reply:
I misunderstood your initial message, I thought it was about the text field of the AlertDialog

I can't use an ImageVector without also providing a contentDescription, I thought it would be better to set a proper one than setting an empty string